Document/0x04e-Testing-Authentication-and-Session-Management.md

```diff
@@ -21,7 +21,7 @@ Web applications commonly use stateful authentication with a random session ID t
 
 As a mobile security tester, you should be familiar with both types of authentication.
 
-### Verifying that Appropriate Authentication is in Place
+### Verifying that Appropriate Authentication is in Place (MSTG‑ARCH‑2 and MSTG‑AUTH‑1)
```

Comment:
Consider the comments in "### Verifying that 2FA is Enforced". I'd move some things from here to over there.

Apart from that we have two sections that might be completely related:
- Verifying that Appropriate Authentication is in Place
- Testing Authentication

I'd extract from these sections whatever is introductory to "authentication" and put it as part of the introduction before/after "Stateful vs. Stateless Authentication". Also consider if "Stateful vs. Stateless Authentication" should be a section at the same level as the "testing" sections. Maybe we prefer to structure it differently.

See more comments in "Testing Authentication".

Reply:
I merged this two now:
- Verifying that Appropriate Authentication is in Place
- Testing Authentication

The "### Stateful vs. Stateless Authentication" Is not a test case but a general overview, which  is needed to clarify on the difference before we can dive into the test cases. There is usually a lot of confusion between stateless and stateful.

The same with "### Verifying that Appropriate Authentication is in Place (MSTG‑ARCH‑2 and MSTG‑AUTH‑1)". This is not really a test case, but describing on a high level how appropriate authentication should look like, as it can be very diverse.